gnrc_sixlowpan_iphc: move assert to a valid position

cgundogan opened the pull request:
The static analyzer was still complaining about `ctx`. I moved the assert out of the not-reachable zone and silenced the static analyzer.

authmillenon commented on sys/net/gnrc/network_layer/sixlowpan/iphc/gnrc_sixlowpan_iphc.c:
```diff
@@ -177,9 +177,10 @@ size_t gnrc_sixlowpan_iphc_decode(gnrc_pktsnip_t *ipv6, gnrc_pktsnip_t *pkt, siz
 
+    /* should be asserted by line 168 anyway */
+    assert(ctx != NULL);
+
```
So context-less compression should not be allowed anymore?

authmillenon commented:
This should be looked at by another person, but IMHO this PR is invalid. `ctx` must only be not NULL if the header has stateful compression activated (SAC and SAM bits set). So asserting that a context is present here will lead to problems when dealing with stateless compression.

cgundogan commented:
@authmillenon I think you are right. I addressed your comment by moving the asserts only to those case blocks where ctx is read and assumed to be `!= NULL`

authmillenon commented:
Nevertheless I would prefer that another person reviews my code :D. @OlegHahm?

cgundogan commented:
squashed

cgundogan commented:
@OlegHahm what's your opinion on this one?

authmillenon commented:
Reviewed this one in more detail (and with some temporal distance to the implementation) again: ACK.
